### PR TITLE
release 3.0.0

### PR DESCRIPTION
**Pull request by traylenator**
release 3.0.0

### CODE REVIEW

**Comment by turnopil** [comment]
Hi, guys! Do you really need this new version? It breaks puppet with old 'nscd' class settings

`Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Nscd]:
  has no parameter named 'hosts_positive_ttl'
  has no parameter named 'passwd_cache'
  has no parameter named 'group_cache'`

**Comment by traylenator** [comment]
Hi @turnopil
For sure it's backwards incompatible.
It makes it massively easier to maintain for all parameters and all OSes however.

See the new README for setting above via class params or hiera.

**Comment by alexjfisher** [comment]
@traylenator Maybe a release summary ticket could be used to emphasise the breaking changes?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,24 @@ All notable changes to this project will be documented in this file.
 Each new release typically also includes the latest modulesync defaults.
 These should not affect the functionality of the module.
 
-## [v2.0.0](https://github.com/voxpupuli/puppet-nscd/tree/v2.0.0) (2019-03-09)
+## [Unreleased](https://github.com/voxpupuli/puppet-nscd/tree/HEAD)
```
Comment (alexjfisher) [inline CHANGELOG.md]: Changelog needs to be generated *after* the metadata is updated.

**Comment by traylenator** [comment]
Unfortunately `bundle exec rake changelog` now fails with:

```
** Execute changelog
Found 6 tags                                                           
rake aborted!dates...
TypeError: no implicit conversion of nil into Array
/home/straylen/GIT/puppet-nscd/vendor/bundle/ruby/2.6.0/bundler/gems/github-changelog-generator-d6836fe504f4/lib/github_changelog_generator/octo_fetcher.rb:289:in `concat'
/home/straylen/GIT/puppet-nscd/vendor/bundle/ruby/2.6.0/bundler/gems/github-changelog-generator-d6836fe504f4/lib/github_changelog_generator/octo_fetcher.rb:289:in `block in commits'
/home/straylen/GIT/puppet-nscd/vendor/bundle/ruby/2.6.0/bundler/gems/github-changelog-generator-d6836fe504f4/lib/github_changelog_generator/octo_fetcher.rb:375:in `iterate_pages'
```

that's 

https://github.com/github-changelog-generator/github-changelog-generator/blob/master/lib/github_changelog_generator/octo_fetcher.rb#L289

**Comment by traylenator** [comment]
Used the --no-http-cache workaround https://github.com/github-changelog-generator/github-changelog-generator/issues/738

**Comment by alexjfisher** [comment]
I *think* the gcg issue might be fixed with a new release of `faraday`.  But also remove your `/tmp/github-changelog-http-cache` directory.

**Comment by traylenator** [comment]
Yep with:

 * faraday (0.17.0)
 * faraday-http-cache (2.0.0)
 * faraday_middleware (0.13.1)

changelog generates okay.

**Comment by alexjfisher** [comment]
@traylenator Do you want to run it one more time?  I've corrected two typos, (one of mine and one of yours).

**Comment by traylenator** [comment]
Weird line endings being injected to changelog from the release-issue

+The previous parameters such as^M

**Comment by traylenator** [comment]
Regenerated - I removed the `^M` manually - ignore until it happens again I'd say.

**Comment by alexjfisher** [comment]
Ah, i think that’s a known issue when using release summaries.

**Comment by alexjfisher** [comment]
https://github.com/github-changelog-generator/github-changelog-generator/issues/715

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,55 @@ All notable changes to this project will be documented in this file.
 Each new release typically also includes the latest modulesync defaults.
 These should not affect the functionality of the module.
 
-## [v2.0.0](https://github.com/voxpupuli/puppet-nscd/tree/v2.0.0) (2019-03-09)
```
Comment (alexjfisher) [inline CHANGELOG.md]: I wonder why this date was wrong previously??

Comment (traylenator) [reply CHANGELOG.md]: Summary 

Currently published with: `v2.0.0 (2019-03-09)`
CHANGELOG now has `v2.0.0 (2019-05-02)


```
2019-02-05  -  a72f8ac (modulesync) [blacksmith] Bump version to 2.0.1-rc0
2019-02-05  -  c3796ad (tag: v2.0.0) Merge pull request #49 from bastelfreak/rel200
2019-07-04  -  012baa8 Merge pull request #50 from voxpupuli/modulesync
2019-07-04  - 8eda5c1 modulesync 2.7.0
2019-09-03  - dfd122d release 2.0.0
```
So a delay between release MR being merged and the actual tagging.

**Comment by traylenator** [comment]
Thanks for the reviews and comments @alexjfisher @bastelfreak @Dan33l and @turnopil

**Pull request merged**